### DISCUSSION TITLE
Arbitrary variants with commas in them

### DISCUSSION THREAD

**Discussion by ben-rogerson** [ben-rogerson/twin.macro#756] [General]
When you use an arbitrary variant with a comma, tailwind only styles with the first selector > [check this demo](https://play.tailwindcss.com/MdB14Ju1bf)
As of Twin v3, the classes you write are converted to tailwind compatible classes (eg: unwrapping variant groups, encoding custom values, replacing theme values etc) before being supplied to tailwind to perform matches.

I don't think this conversion should be silent in twin, so I've made twin throw a helpful error instead.

Also, the reason we can use commas within a single arbitrary variant is because twin pre-encodes the commas before the matches - A trick that only works when there isn't additional variants on the class.

**Comment by CadenP** [comment]
A workaround for some situations with multiple selectors is to use `:is(.first, .second)`: [demo](https://play.tailwindcss.com/3Drqr0oawu)

[MDN Docs for :is()](https://developer.mozilla.org/en-US/docs/Web/CSS/:is)
